### PR TITLE
Check runtime benchmarks in CI

### PR DESCRIPTION
**Pull request by Kobzol**
We cannot run the benchmarks in CI yet because of `perf` issues (I'll resolve it after https://github.com/rust-lang/rustc-perf/pull/1463), but we can already check if they compile, at least.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -116,6 +116,40 @@ jobs:
...
+  test_runtime_benchmarks:
+    name: Test runtime benchmarks
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the source code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 1
+
+      - name: Install latest nightly
```
Comment (nnethercote) [inline .github/workflows/ci.yml:128]: Why install nightly? For the compile-time benchmarks we only install beta.

File: ci/check-runtime-benchmarks.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+set -eE -x;
+
+bash -c "while true; do sleep 30; echo \$(date) - running ...; done" &
+PING_LOOP_PID=$!
+trap 'kill $PING_LOOP_PID' ERR 1 2 3 6
+
+# Check if the runtime benchmarks can be compiled
```
Comment (nnethercote) [inline ci/check-runtime-benchmarks.sh:9]:
```suggestion
# Check if the runtime benchmarks can be compiled.
```

Comment (nnethercote) [reply ci/check-runtime-benchmarks.sh:9]: Is there a plan to do more here eventually? Is it worth mentioning in a comment?

File: ci/check-runtime-benchmarks.sh [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+cd collector/runtime-benchmarks
```
Comment (nnethercote) [inline ci/check-runtime-benchmarks.sh:10]: While here, is it worth renaming `ci/check-benchmarks.sh` as `ci/check-compile-benchmarks.sh`?

**Review by nnethercote — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Kobzol) [reply .github/workflows/ci.yml:128]: The function that we use to gather a toolchain for runtime benchmarks tries to find a nightly toolchain using rustup, if it's not available, it fails.. I'm not sure if it's required, but I didn't want to change existing code. But it's not needed for check, so I'll remove it for now.

File: ci/check-runtime-benchmarks.sh (hunk 2 above)
Comment (Kobzol) [reply ci/check-runtime-benchmarks.sh:9]: I added a comment for that.

File: ci/check-runtime-benchmarks.sh (hunk 3 above)
Comment (Kobzol) [reply ci/check-runtime-benchmarks.sh:10]: Done.

**Pull request merged**